src/schema/v2/me/collectArtistsMutation.ts

```diff
@@ -0,0 +1,92 @@
+import {
+  GraphQLList,
+  GraphQLNonNull,
+  GraphQLObjectType,
+  GraphQLString,
+  GraphQLUnionType,
+} from "graphql"
+import { mutationWithClientMutationId } from "graphql-relay"
+import { uniq } from "lodash"
+import { ResolverContext } from "types/graphql"
+import { UserInterest, userInterestType } from "../userInterests"
+
+interface Input {
+  artistIDs: string[]
+}
+
+const SuccessType = new GraphQLObjectType<any, ResolverContext>({
+  name: "collectArtistsSuccess",
+  fields: () => ({
+    userInterests: {
+      type: new GraphQLNonNull(new GraphQLList(userInterestType)),
+      resolve: (userInterests) => userInterests,
+    },
+  }),
+})
+
+const FailureType = new GraphQLObjectType<any, ResolverContext>({
+  name: "collectArtistsFailure",
+  fields: () => ({
+    mutationError: {
+      type: new GraphQLNonNull(GraphQLString),
+      resolve: (err) => err,
+    },
+  }),
+})
+
+const UserInterestsOrErrorType = new GraphQLUnionType({
+  name: "collectArtistsResponseOrError",
+  types: [SuccessType, FailureType],
+  resolveType: (data) =>
+    data._type === "GravityMutationError" ? FailureType : SuccessType,
+})
+
+export const collectArtistsMutation = mutationWithClientMutationId<
+  Input,
+  UserInterest[] | null,
+  ResolverContext
+>({
+  name: "CollectArtistsMutation",
+  description: "Collect Multiple Artists",
+  inputFields: {
+    artistIDs: {
+      type: new GraphQLNonNull(
+        new GraphQLList(new GraphQLNonNull(GraphQLString))
+      ),
+    },
+  },
+  outputFields: {
+    userInterestsOrError: {
+      type: UserInterestsOrErrorType,
+      description: "on success: returns the user interest",
+      resolve: (result) => result,
+    },
+  },
+  mutateAndGetPayload: async (args, { meCreateUserInterestLoader }) => {
+    if (!meCreateUserInterestLoader) {
+      throw new Error("You need to be signed in to perform this action")
+    }
+    const uniqueArtistIDs = uniq(args.artistIDs)
+
+    try {
+      const userInterests = await Promise.all(
+        uniqueArtistIDs.map((artistID) => {
+          const gravityPayload = {
+            interest_id: artistID,
+            interest_type: "Artist",
+            category: "collected_before",
+          }
+          return meCreateUserInterestLoader(gravityPayload)
+        })
+      )
+
+      return userInterests
+    } catch (error) {
+      if ("body" in (error as any)) {
```

Comment:
I _think_ this can just re-use the standard handling for errors, a la https://github.com/artsy/metaphysics/blob/e1d3950fc75966e218d53660633918452c7cae2f/src/schema/v2/Feature/CreateFeatureMutation.ts#L110-L115

If there's something about Gravity's response or MP parsing that seems to not play nicely with that - let me know and I'll take a look and fix it up, since we should try to expose these errors in a standard/vanilla way.